lib/adapters/parsers/csv.js

```diff
@@ -22,10 +28,19 @@ module.exports = base.extend({
             });
 
             csvStream.on('data', function(pt){
-                buffer.push(pt);
-                if (buffer.length >= self.limit) {
-                    emit(buffer);
-                    buffer = [];
+                self.totalParsed++;
+
+                if (self.stopAt !== -1 && self.totalParsed > self.stopAt) {
```

Comment:
Let's have the unoptimized `stopAt` be `Infinity` so we don't need the extra `!== -1` check.